libs/ActivityKit/source/ActivityKit.cpp

```diff
@@ -0,0 +1,31 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ActivityKit.h"
+
+using namespace leka;
+
+void ActivityKit::registerActivities(std::unordered_map<MagicCard, interface::Activity> const &activities)
+{
+	_activities = activities;
+}
+
+void ActivityKit::start(const MagicCard &card)
+{
+	stop();
+
+	_activity = &_activities.at(card);
+
+	if (_activity == nullptr) {
+		return;
+	}
```

Comment:
ce check là est inutile parce qu'avec le map ça n'arrivera pas.

avec `map.at(key)` tu as un bound check, donc si la clé n'existe pas tu auras une erreur et ça plantera.

tu peux faire

```suggestion
	stop();
	
	if (auto activity_not_available = !_activities.contains(card); activity_not_available) {
		_current_activity == nullptr;
		return;
	}
	
	_current_activity = _activities.at(card);
	
	_activity->start();
}
```